### PR TITLE
Add io.netty.handler.codec.http.DefaultHttpResponse to Netty Response Splitting Detection

### PR DESCRIPTION
**Pull request by JLLeitschuh**
Related: #2185
Related: https://github.com/github/security-lab/issues/22

### CODE REVIEW

**Comment by aschackmull** [comment]
CI triggered.

File: java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql [hunk 1]
```diff
@@ -13,8 +13,21 @@
...
-where
-  new.getConstructedType().hasQualifiedName("io.netty.handler.codec.http", "DefaultHttpHeaders") and
-  new.getArgument(0).getProperExpr().(BooleanLiteral).getBooleanValue() = false
-select new, "Response-splitting vulnerability due to verification being disabled."
+abstract private class InsecureNettyObjectCreation extends ClassInstanceExpr { }
+
+private class InsecureDefaultHttpHeadersClassInstantiation extends InsecureNettyObjectCreation {
+  InsecureDefaultHttpHeadersClassInstantiation() {
+    getConstructedType().hasQualifiedName("io.netty.handler.codec.http", "DefaultHttpHeaders") and
+    getArgument(0).getProperExpr().(BooleanLiteral).getBooleanValue() = false
```
Comment (JLLeitschuh) [inline java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql]:
```suggestion
    getArgument(0).getProperExpr().(CompileTimeConstantExpr).getBooleanValue() = false
```

Would this still work or would this be wrong?
If this works, I should do the same thing below as well!

Comment (aschackmull) [reply java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql]: That works, yes, and should be a strict improvement.  With that you can even simplify and drop the `.getProperExpr()` as that is included in `CompileTimeConstantExpr`.

Comment (aschackmull) [reply java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql]:
```suggestion
    getArgument(0).(CompileTimeConstantExpr).getBooleanValue() = false
```

File: java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql [hunk 2]
```diff
@@ -13,8 +13,21 @@
...
+  }
+}
+
+private class InsecureDefaultHttpResponseClassInstantiation extends InsecureNettyObjectCreation {
+  InsecureDefaultHttpResponseClassInstantiation() {
+    getConstructedType().hasQualifiedName("io.netty.handler.codec.http", "DefaultHttpResponse") and
+    getArgument(2).getProperExpr().(BooleanLiteral).getBooleanValue() = false
```
Comment (JLLeitschuh) [inline java/ql/src/Security/CWE/CWE-113/NettyResponseSplitting.ql]:
```suggestion
    getArgument(2).(CompileTimeConstantExpr).getBooleanValue() = false
```

**Comment by JLLeitschuh** [comment]
Thanks @aschackmull!!!

**Comment by JLLeitschuh** [comment]
Woot!!!

**Pull request merged**